enhancements/cluster-logging/gcp-external-account-auth.md

```diff
@@ -0,0 +1,201 @@
+
+Current GCP authentication in Vector relies on service account credentials, which require 
+static key management. WIF eliminates the need for long-lived credentials, enabling more secure, 
+short-lived authentication via OpenShift service account tokens.
+
+### User Stories
+
+* As an OpenShift administrator, I want Vector to authenticate with GCP using Workload Identity Federation, 
+removing the need for static service account keys. 
+* As an engineer, I want Vector to detect the correct authentication method based on the credentials file 
+format, utilizing the correct credentials chain form the sdk, and ensuring compatibility with 
+both service accounts and external accounts (WIF).
+
+
+### Goals
+
+* Modify the GCP module in vector to support external_account credentials. 
+* Introduce new structs (ExternalCredentials, CredentialsType) to handle the updated credential format. 
+* Serialize the configuration file into usable data 
+* Implement the full OIDC-based authentication flow. 
+* Ensure compatibility with existing service account authentication
+
+
+### Non-Goals
+
+* Modifying the OpenShift gcp output authentication model. 
+* Supporting authentication methods outside of Google Cloud.
+
+
+## Proposal
+
+To integrate WIF, Vector’s authentication logic will be updated to support external_account 
+credentials in addition to the existing service_account credentials. The process will:
+
+* Determine the authentication type by reading the credentials JSON file. 
+* Extract the OpenShift Service Account token from the projected volume. 
+* Exchange the OpenShift token for a Google Identity Token using Google STS. 
+* Exchange the Identity Token for a short-lived Access Token via Google IAM impersonation. 
+* Ensure the existing service account authentication remains functional.
+
+### Workflow Description
+### API Extensions
+The existing credentials key and secret works for our purpose, although we should aim to eventually  
+align with our cloudwatch and future Azure specs which are most likely going to be configMaps:
+```yaml
+ googleCloudLogging:
+   authentication:
+     credentials:
+       key: google-ext-account-creds.json
+       secretName: gcp-secret
+   logId : my-gcp
+...
+```
+### Topology Considerations
+#### Hypershift / Hosted Control Planes
+#### Standalone Clusters
+#### Single-node Deployments or MicroShift
+### Implementation Details/Notes/Constraints
+
+1. Modify Authentication to Support external_account  
+   * Update authentication logic to check for external_account type.
+   * Add a CredentialsType struct to differentiate between service_account and external_account.
+2. Introduce ExternalCredentials Struct   
+   * Create a new struct to handle external_account authentication fields.
+   * Ensure compatibility with Google’s OIDC-based credential format.
+3. Implement the WIF Authentication Flow   
+   * Step 1: Read the OpenShift service account token from the path specified in the credentials file   
```

Comment:
maybe we should mention specifically this is the projected service account token.

I will also mention this path is up to us to manage, regardless of what the file says.  Maybe we need to copy and modify the content if the lib will be reading the path directly from the file.  The path is from the collector's perspective.  The admin needs no control or input of where the token lives; it is immaterial to them.  Additionally, it is very much possible the projected SA path  is not the same as what is defined in the credentials file

Reply:
This is mainly for upstream vector functionality.    For the user, this is a required value that must be set to the path of their SA token.  It is the same as Loki and other apps have implemented.   

Too much abstraction is not a good thing.  The user should know that tokens are located at a specific path in an openshift cluster, if they are creating these files and setting all this up.   Knowledge of the OIDC setup and configuration is assumed and I think its fairly standard to require a specific value for the token location.       I've already created the user docs referencing this, and it will be required.   We could always include validation?    Otherwise vector debug will actually state:   "token not found at location"